openedx/core/djangoapps/content/learning_sequences/api/tests/test_outlines.py

```diff
@@ -1279,6 +1288,405 @@ def test_private_course_outline(self):
                         'Sequences should be accessible to enrolled, staff users for a public_outline course'
 
 
+@ddt.ddt
+class EnrollmentTrackPartitionGroupsTestCase(OutlineProcessorTestCase):  # lint-amnesty, pylint: disable=missing-class-docstring
+    """Tests for enrollment track partitions outline processor that affect outlines"""
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        cls.visibility = VisibilityData(
+            hide_from_toc=False,
+            visible_to_staff_only=False
+        )
+
+    def _add_course_mode(
```

Comment:
My personal policy is to only require them in public API functions and to otherwise just sprinkle them where they seem helpful. They are sadly not as useful because we're a long way from being able to run mypy on edx-platform right now. (I took an hour detour on this at one point and gave up on it as something that will require a lot more work).